lpc55_sign_bin/src/main.rs

```diff
@@ -199,33 +199,66 @@ enum Command {
         beacon: u16,
     },
+    /// Generates a TOML file encapsulating certificate settings, which can then
+    /// be passed to the --cert-cfg option for other subcommands.
+    GenCertCfg {
+        /// Path to private key, if available.
+        #[clap(long)]
+        private_key: Option<PathBuf>,
+        /// Path to root cert; can be repeated.
+        #[clap(long)]
+        root_cert: Vec<PathBuf>,
+        /// Path to signing cert; can be repeated.
+        #[clap(long)]
+        signing_cert: Vec<PathBuf>,
+    },
 }
 
 #[derive(Debug, Parser)]
 struct CertArgs {
-    #[clap(long, requires = "root_cert")]
+    /// Path to private key file, if available. Not all operations require this.
+    ///
+    /// Cannot be combined with the --cert-cfg option.
+    #[clap(long)]
     private_key: Option<PathBuf>,
-    #[clap(long, requires = "private_key")]
-    root_cert: Option<PathBuf>,
-
+    /// Path to root certificate. Can be repeated up to 4 times.
+    ///
+    /// This interface will also reuse the root certificates as the signing
+    /// certificates. To override signing certificates, see the --cert-cfg
+    /// option.
+    ///
+    /// Cannot be combined with the --cert-cfg option.
+    #[clap(long)]
+    root_cert: Vec<PathBuf>,
+
+    /// Path to a TOML file specifying the cert configuration. This file can
+    /// contain three top-level keys, all optional. `private-key` gives the path
+    /// to a private key on disk, if available. `root-certs` should be an array
+    /// of strings giving the absolute paths to the root certificates.
+    /// `signing-certs` should be an array of strings giving the absolute paths
+    /// to signing certificates.
+    ///
+    /// This is intended to avoid having to pass all that on the command line
+    /// every time.
+    ///
+    /// See the `gen-cert-cfg` subcommand to generate this file automatically.
+    ///
+    /// Cannot be combined with either the --private-key or --root-cert options.
     #[clap(long, conflicts_with_all = ["private_key", "root_cert"])]
     cert_cfg: Option<PathBuf>,
 }
 
 impl CertArgs {
     fn try_into_config(self) -> Result<CertConfig> {
-        if (self.private_key.is_none() || self.root_cert.is_none()) && self.cert_cfg.is_none() {
-            bail!("must provide either root-cert + private-key, or cert-cfg")
-        } else if let Some(s) = self.cert_cfg {
+        if let Some(s) = self.cert_cfg {
             let cfg_contents = std::fs::read_to_string(s)?;
             let cfg = toml::from_str(&cfg_contents)?;
             Ok(cfg)
         } else {
-            let root = self.root_cert.unwrap();
             Ok(CertConfig {
-                private_key: self.private_key.unwrap(),
-                root_certs: vec![root.clone()],
-                signing_certs: vec![root],
+                private_key: self.private_key,
+                root_certs: self.root_cert.clone(),
+                signing_certs: self.root_cert,
```

Comment:
This doesn't seem correct to me: `root_certs` is the set of four root keys, but `signing_certs` is meant to be a certificate chain `[root cert, signing cert]`

Reply:
Ah, so the original code was doing that implicitly by only accepting one root cert, you think?